inference_overview.md

```diff
@@ -0,0 +1,449 @@
+features will need to run inside the TE, and cannot run outside the TEE
+boundary.
+
+### Unified models
+
+A **unified** approach to such a prediction could run a model that takes
+contextual, ad and user features as input, and makes a prediction, running the
+entire model inside the TEE. B&A will support unified models directly through
+the [two ways][19] of running inference (embedded and external models) described
+above. In the unified approach, the entire model will need to run inside the
+TEE, since it uses private features. These models will tend to be larger. Both
+these factors can increase the cost and latency of running such models.
+
+### Factorized models
+
+**Model factorization** is a technique that makes it possible to break a single
+model into multiple pieces, and then combine those pieces into a prediction. In
+B&A-supported use cases, models often make use of three kinds of data: user
+data, contextual data, and ad data.
+
+As described above, in the unified case, a single model is trained on all three
+kinds of data. In the factorized case, you could break the model up into
+multiple pieces. The model that includes the sensitive user data needs to be
+executed within the trust boundary, on the buyer's bidding services.
+
+This makes the following design possible:
+
+1. Break the model up into a private piece (the sensitive user data) and one or
+  more non-private pieces (the contextual and ad data).
+1. Optionally, inferences from some or all of the non-private pieces can be
+  passed in as arguments to a [User Defined Function][1] (UDF), e.g.
+  `generateBid()`. For example, contextual embeddings can be computed along the
+  contextual path, returned to the seller's frontend in the contextual response,
+  and then can be passed into the B&A auction as part of the `buyer_signals`
+  constructed by the seller when they initiate the Auction.
+1. Optionally, ad techs can create models for non-private pieces (such as ads
+  features) ahead of time, then materialize embeddings from those models into
+  the ad retrieval server and fetch these embeddings at runtime.
+1. To make a prediction during a UDF, combine private embeddings from the
+  inference service running in a TEE with non-private embeddings from UDF
+  function arguments or the ad retrieval server store with an operation like a
+  dot product. This is the final prediction.
+
+The following figure shows a factorized pCTR model broken into three parts,
+contextual, user and ads towers. Only the User tower needs to run inside the
+TEE. The contextual and ad towers can run outside the TEE since they do not rely
+on private features.
+
+<figure id = "image-1">
+  <img src = "images/inference_overview_fig1.png"
+  width = "100%"
+  alt = "Factorized pCTR model">
+  <figcaption><b>Figure 1.</b> Factorized pCTR Model </figcaption>
+</figure>
+
+Factorized models are useful because loading and executing large models inside
```

Comment:
Nit: Can you add a line break right after the figure? The same applies to figure #2 as well.

Reply:
Done.

Reply:
Did it work? From https://github.com/privacysandbox/protected-auction-services-docs/blob/55be73574890870bb2d119402bbd5f4e7faea684/inference_overview.md#inference-with-externally-provided-models , I do not see a blank line.

Reply:
trying with an additional blank line to see if it works. If not, i can update this after submission (since I can edit inline and see how it looks).

Reply:
OK, so extra blank line didn't work. I can follow up post submit to figure out whats the best way to do this that works with the github rendering.